docs/sorting-algorithms/insertion-sort.mdx

```diff
@@ -0,0 +1,194 @@
+
+        /* Here the elements in b/w arrary[0 to i-1]
+        which are greater than target are moved
+        ahead by 1 position each*/
+        while (j >= 0 && array[j] > target) 
+        {  
+            array[j + 1] = array[j];  
+            j = j - 1;  
+        }  
+        array[j + 1] = target;  
+    }
+    cout<<"After Insertion sort: \n"; 
+    display(array, size);//Using dispaly function to print sorted array. 
+    return 0; 
+}
+```
+
+</TabItem>
+
+<TabItem value="py">
+
+```python
+"""
+This is a pure Python implementation of the insertion sort algorithm
+"""
+
+""" Implementation of the insertion sort algorithm in Python
+    :parameter array: A mutable ordered collection with comparable items inside
+    
+    :return : the same collection ordered by ascending
+    Examples:
+    >>> insertion_sort([0, 5, 3, 2, 2])
+    [0, 2, 2, 3, 5]
+    >>> insertion_sort([])
+    []
+    >>> insertion_sort([-2, -5, -45])
+    [-45, -5, -2]
+"""
+def insertion_sort(array):
+    # Loop from the second element of the array until
+    # the last element
+    for i in range(1, len(array)):
+        # This is the element we want to position in its
+        # correct place
+        key_item = array[i]
+
+        # Initialize the variable that will be used to
+        # find the correct position of the element referenced
+        # by `key_item`
+        j = i - 1
+
+        # Run through the list of items (the left
+        # portion of the array) and find the correct position
+        # of the element referenced by `key_item`. Do this only
+        # if `key_item` is smaller than its adjacent values.
+        while j >= 0 and array[j] > key_item:
+            # Shift the value one position to the left
+            # and reposition j to point to the next element
+            # (from right to left)
+            array[j + 1] = array[j]
+            j -= 1
+
+        # When you finish shifting the elements, you can position
+        # `key_item` in its correct location
+        array[j + 1] = key_item
+
+    return array
+
+
+
+if __name__ == "__main__":
+
+
+    user_input = list(map(int, input("Enter a sequence of comma seprated numbers:\n").split(',')))
+    
+    # calling the insertion_sort function
+    sorted_collection =insertion_sort(user_input)
+    print(sorted_collection)
+```
+
+</TabItem>
+</Tabs>
+
+## Sample I/O
+
+### Sample Input
+
+```bash
+5
+3
+1
+9
+8
+2
+4
+7
+```
+
+### Sample Output
+
+```bash
+Before Insertion sort: 
+5   3   1   9   8   2   4   7
+After Insertion sort: 
+1   2   3   4   5   7   8   9
+```
+
+## Complexity Analysis
+
+- `n` is the number of elements in the array.
+
+- Time Complexity:
+
+  - **Best Case**: O (n)
+  - **Average Case**: O (n<sup>2</sup>)
+  - **Worst Case**: O (n<sup>2</sup>)
+
+- Space Complexity: O (1)
+
+## Credits
+
+- [Rajiv Singh](https://github.com/iamrajiv) for C++ implementation and [Ricardo Prins](https://github.com/ricardoprins) for python implementation.
```

Comment:
- Use: `Rajiv Singh for the C++ implementation`. Same for Python
- It was Keshav who did the Python implementation. Here is the PR: https://github.com/TesseractCoding/NeoAlgo/pull/29

Reply:
@HarshCasper so we only need names and no links to their GitHub correct?

Reply:
@HarshCasper so we only need names and no links to their GitHub correct?

Reply:
Links to GitHub would be alright. You can however skip them if you want or if the information is not available